shared/src/main/scala/coop/rchain/shared/ByteStringOps.scala

```diff
@@ -0,0 +1,116 @@
+package coop.rchain.shared
+
+import java.io._
+import java.util.concurrent.{Callable, Executors}
+
+import com.google.protobuf.ByteString
+import org.apache.commons.compress.compressors.lz4.FramedLZ4CompressorOutputStream.Parameters
+import org.apache.commons.compress.compressors.lz4.{
+  FramedLZ4CompressorInputStream,
+  FramedLZ4CompressorOutputStream
+}
+
+import scala.util.{Failure, Success, Try}
+
+object ByteStringOps {
+
+  implicit class RichByteString(bs: ByteString) {
+    def compress: ByteString = compressLZ4(bs)
+
+    def decompress: Option[ByteString] = decompressLZ4(bs)
+  }
+
+  private def compressLZ4(bs: ByteString): ByteString = {
+    val pos = new PipedOutputStream()
+    val pis = new PipedInputStream(pos)
+    val is  = bs.newInput()
+
+    // Writes compressed data into `pos`
+    val producer = new Runnable {
+      override def run(): Unit = {
+        var lz4os: FramedLZ4CompressorOutputStream = null
+        try {
+          lz4os = new FramedLZ4CompressorOutputStream(pos, Parameters.DEFAULT)
```

Comment:
why do you use a var and not initialize lz4os with `new FramedLZ4CompressorOutputStream(...)` in the first place?

Reply:
This is the most likely java'like copy-paste. There is an try-catch part, which could be exchanged for `Try`

Reply:
I wish I could have just copy-pasted this from somewhere ;)

I'm using a var because I want to close `lz4os` in the finally block in the case that `new FramedLZ4CompressorOutputStream(...)` explodes.

Reply:
if the c-tor explodes there is nothing to close

Reply:
Not necessarily.
For example `FramedLZ4CompressorOutputStream.close()` closes the output stream that gets passed to the constructor.
If the constructor explodes after the output stream is set to `this.out`, there is something to close.
After thinking about this it seems like line 44 and 92 are not needed.